src/Storages/PartitionStrategy.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <Interpreters/ExpressionActions.h>
+#include <Parsers/IAST_fwd.h>
+
+#include <Processors/Chunk.h>
+
+namespace DB
+{
+
+struct PartitionStrategy
```

Comment:
Let's add a general comment above this struct about what is this class. Same for `PartitionStrategyFactory`.